From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Sun, 19 Apr 2026 18:42:01 -0500
Subject: [PATCH 20/46] 
 0025-winewayland-Do-not-process-motion-events-when-there-.patch

---
 dlls/winewayland.drv/wayland_pointer.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/dlls/winewayland.drv/wayland_pointer.c b/dlls/winewayland.drv/wayland_pointer.c
index 86316441c6a..7e3af8e7ddc 100644
--- a/dlls/winewayland.drv/wayland_pointer.c
+++ b/dlls/winewayland.drv/wayland_pointer.c
@@ -329,6 +329,13 @@ static void pointer_handle_frame(void *data, struct wl_pointer *wl_pointer)
 
     pthread_mutex_lock(&pointer->mutex);
 
+    /* cannot handle any new motion events when a warp is pending */
+    if (pointer->pending_warp)
+    {
+        frame->flags &= ~(WAYLAND_POINTER_FRAME_ABSOLUTE
+                          | WAYLAND_POINTER_FRAME_RELATIVE);
+    }
+
     input.mi.dwFlags = MOUSEEVENTF_ABSOLUTE | MOUSEEVENTF_MOVE;
 
     if (frame->flags & WAYLAND_POINTER_FRAME_ABSOLUTE)
